From f2273101c21bc59a390379e182e53cd4f07a7e71 Mon Sep 17 00:00:00 2001
From: Benjamin Franzke <benjaminfranzke@googlemail.com>
Date: Thu, 19 Feb 2015 20:47:28 +0100
Subject: machined: use x-machine-unix prefix for the container bus on dbus1

This fixes "machinectl login" on systems configured with --disable-kdbus.

The error was:
machinectl login foo
Failed to get machine PTY: Input/output error

diff --git a/src/machine/machine-dbus.c b/src/machine/machine-dbus.c
index 15c9159..9e78a67 100644
--- a/src/machine/machine-dbus.c
+++ b/src/machine/machine-dbus.c
@@ -511,7 +511,7 @@ int bus_machine_method_open_login(sd_bus *bus, sd_bus_message *message, void *us
 #ifdef ENABLE_KDBUS
         asprintf(&container_bus->address, "x-machine-kernel:pid=" PID_FMT ";x-machine-unix:pid=" PID_FMT, m->leader, m->leader);
 #else
-        asprintf(&container_bus->address, "x-machine-kernel:pid=" PID_FMT, m->leader);
+        asprintf(&container_bus->address, "x-machine-unix:pid=" PID_FMT, m->leader);
 #endif
         if (!container_bus->address)
                 return log_oom();
-- 
cgit v0.10.2


